sphinx/builders/html.py

```diff
@@ -785,7 +785,7 @@ def pathto(otheruri, resource=False, baseuri=default_baseuri):
                 return otheruri
             elif not resource:
                 otheruri = self.get_target_uri(otheruri)
-            uri = relative_uri(baseuri, otheruri) or '#'
+            uri = relative_uri(baseuri, otheruri) or baseuri
```

Comment:
This also effects to HTML builders. With chrome, this causes page reloading. It is not so bad, but unkind for me. What do you think?
It would be nice if epub builder overrides this locally.
I believe the `html-page-context` event is good to do that.